cmd/argocd/commands/app.go

```diff
@@ -1617,3 +1619,67 @@ func NewApplicationPatchCommand(clientOpts *argocdclient.ClientOptions) *cobra.C
 	command.Flags().StringVar(&patch, "patch", "", "Patch")
 	return &command
 }
+
+func NewApplicationPatchResourceCommand(clientOpts *argocdclient.ClientOptions) *cobra.Command {
+	var patch string
+	var patchType string
+	var resourceName string
+	var namespace string
+	var kind string
+	var group string
+	command := cobra.Command{
+		Use:   "patch-resource APPNAME",
+		Short: "Patch resource in an application",
+		Run: func(c *cobra.Command, args []string) {
+			if len(args) != 1 {
+				c.HelpFunc()(c, args)
+				os.Exit(1)
+			}
+			appName := args[0]
+
+			conn, appIf := argocdclient.NewClientOrDie(clientOpts).NewApplicationClientOrDie()
+			defer util.Close(conn)
+			ctx := context.Background()
+			resources, err := appIf.ManagedResources(ctx, &application.ResourcesQuery{ApplicationName: &appName})
+			errors.CheckError(err)
+			liveObjs, err := liveObjects(resources.Items)
+			errors.CheckError(err)
+			for _, obj := range liveObjs {
+				gvk := obj.GroupVersionKind()
+				if kind != "" && kind != gvk.Kind {
+					continue
+				}
+				if group != "" && kind != gvk.Group {
+					continue
+				}
+				if namespace != "" && namespace != obj.GetNamespace() {
+					continue
+				}
+				if resourceName == obj.GetName() {
+					_, err := appIf.PatchResource(ctx, &application.ApplicationResourcePatchRequest{
+						Name:         &appName,
+						Namespace:    obj.GetNamespace(),
+						ResourceName: resourceName,
+						Version:      gvk.Version,
+						Group:        gvk.Group,
+						Kind:         gvk.Kind,
+						Patch:        patch,
+						PatchType:    patchType,
+					})
+					errors.CheckError(err)
+					break
+				}
+			}
+		},
```

Comment:
If we fail to find a resource, we should also error.

Reply:
fixed